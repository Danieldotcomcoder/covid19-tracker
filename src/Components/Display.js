import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Item from './Item';

const Display = ({ items = [] }) => (
  <ul className="Home-grid">
    {items.map(({ name, confirmed }) => (
      <li key={name} className="Home-grid-item">
        <Link to={`/country/${name}`}>
          <Item confirmed={confirmed} name={name} />
        </Link>
      </li>
    ))}
  </ul>
);

Display.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape(Item.propTypes)).isRequired,
};

export default Display;
