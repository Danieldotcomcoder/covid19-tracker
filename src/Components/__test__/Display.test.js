import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../Redux/configureStore';
import Display from '../Display';

it('Renders Display Component', () => {
  const Displaypage = renderer.create(
    <Provider store={store}>
      <Display />
      ,
    </Provider>,
  ).toJSON();
  expect(Displaypage).toMatchSnapshot();
});
