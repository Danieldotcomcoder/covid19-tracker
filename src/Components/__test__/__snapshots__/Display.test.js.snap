// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Display Component 1`] = `
Array [
  <ul
    className="grid"
  />,
  ",",
]
`;
